src/test/java/org/openrewrite/staticanalysis/RemoveUnusedPrivateMethodsTest.java

```diff
@@ -103,6 +103,488 @@ private void dontRemove2() {
+                            // the nest host class is always in the nest
+                            String className = javaClass.getClassName();
+                            nestMateClassNames.add(className);
+                        }
+                    }
+                    return nestMateClassNames;
+                }
+
+                private static boolean hasJava11OrAboveClassVersion(JavaClass... javaClasses) {
+                    for (JavaClass javaClass : javaClasses) {
+                        int javaClassVersion = javaClass.getMajor();
+                        if (javaClassVersion < JAVA_11_CLASS_VERSION) {
+                            return false;
+                        }
+                    }
+                    return true;
+                }
+
+                @CheckForNull
+                private static String[] getNestMemberClassNames(JavaClass javaClass) {
+                    Attribute[] sourceAttributes = javaClass.getAttributes();
+                    for (Attribute sourceAttribute : sourceAttributes) {
+                        if (sourceAttribute instanceof NestMembers) {
+                            NestMembers nestMembersAttribute = (NestMembers) sourceAttribute;
+                            return nestMembersAttribute.getClassNames();
+                        }
+                    }
+                    return null;
+                }
+
+                @CheckForNull
+                private static String getHostClassName(JavaClass javaClass) {
+                    Attribute[] attributes = javaClass.getAttributes();
+                    for (Attribute attribute : attributes) {
+                        if (attribute instanceof NestHost) {
+                            NestHost nestHostAttribute = (NestHost) attribute;
+                            int targetHostClassIndex = nestHostAttribute.getHostClassIndex();
+                            ConstantPool constantPool = nestHostAttribute.getConstantPool();
+                            return constantPool.getConstantString(targetHostClassIndex, Const.CONSTANT_Class);
+                        }
+                    }
+                    return null;
+                }
+
+                public static List<JavaClass> getHostClasses(JavaClass javaClass) throws ClassNotFoundException {
+                    List<JavaClass> list = new ArrayList<>();
+                    if (javaClass != null) {
+                        String hostClassName = getHostClassName(javaClass);
+                        if (hostClassName != null) {
+                            JavaClass hostClass = Repository.lookupClass(hostClassName);
+                            list.add(hostClass);
+                            list.addAll(getHostClasses(hostClass));
+                        }
+                    }
+                    return list;
+                }
+            }
+            """
+          )
+        );
+    }
+
```

Comment:
Can we reduce this down to just the relevant bits? We surely don't need 400 lines just to replicate an issue. A lot of this will not be relevant, but still needs to be evaluated whenever we try to mentally parse this test.

Reply:
yes first idea was to keep code like it is due to licence, will try to shrink as well.